src/main/java/gregtech/api/metatileentity/interfaces/IHasWorldObjectAndCoords.java

```diff
@@ -18,6 +21,11 @@ default boolean isClientSide() {
         return world() != null && world().isRemote;
     }
 
+    @Nullable
+    TileEntity getNeighbor(EnumFacing facing);
+
+    void onNeighborChanged(EnumFacing facing);
```

Comment:
These should be pulled into a separate interface. Also, it would be better to implement these methods further down in the inheritance tree if possible. I think this class is probably too high up for it.